api/src/play_asset_pack_manager.gd

```diff
@@ -24,9 +24,24 @@
 # ##############################################################################
 extends Node
 
+# -----------------------------------------------------------------------------
+# Emits state_updated(pack_name, state) global signal upon any asset pack's 
+# state update.
+# 	pack_name : String name of the pack
+#	state: PlayAssetPackState object of the updated state
+# -----------------------------------------------------------------------------
+signal state_updated(pack_name, state)
+
 var _plugin_singleton : Object
 var _request_tracker : PlayAssetDeliveryRequestTracker
 
+# Dictionary that stores the mapping of pack_name to relevant Request objects	
+var _asset_pack_to_request_map : Dictionary	
+var _play_asset_pack_manager_mutex : Mutex	
+var _asset_pack_state_cache : Dictionary
```

Comment:
Could we use the AssetPackStates within the requests themselves instead of maintaining a separate cache? If not, add a comment here explaining why this cache is necessary.

Reply:
This cache is needed since the request object's state  is updated on main thread, and we are guaranteed to get most updated state from it.